effort to link with gsl installed from system
to be applied after the as-needed patch
bicatali@gentoo.org - 06/2013

diff -Nur astrometry.net-0.43.as-needed/blind/Makefile astrometry.net-0.43/blind/Makefile
--- astrometry.net-0.43.as-needed/blind/Makefile	2013-06-13 15:54:15.915430597 -0700
+++ astrometry.net-0.43/blind/Makefile	2013-06-13 17:09:54.247653801 -0700
@@ -56,7 +56,7 @@
 LDFLAGS += -fPIC
 
 LDLIBS := $(LDLIBS_DEF)
-LDLIBS += -lm
+LDLIBS += -lm $(GSL_LIB)
 
 CFLAGS += $(CFLAGS_DEF)
 CFLAGS += $(ANUTILS_INC)
@@ -113,7 +113,7 @@
 PROSPECTUS := quadidx codeprojections quadscales quadsperstar \
 	quadcenters startree2rdls
 
-ANLIBS := $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+ANLIBS := $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 BACKEND_LIB := libbackend.a
 
@@ -138,9 +138,9 @@
 
 SHAREDLIBFLAGS := $(SHAREDLIBFLAGS_DEF)
 
-$(BACKEND_SO): $(BACKEND_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
-		$(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(SHAREDLIBFLAGS) $^ -o $@
+$(BACKEND_SO): $(BACKEND_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(LIBKD_LIB) \
+		$(QFITS_LIB) 
+	$(CC) $(LDFLAGS) $(SHAREDLIBFLAGS) $^ $(GSL_LIB) -o $@
 
 # old and miscellaneous executables that aren't part of the pipeline.
 OLDEXECS := plotquads rawstartree checkquads
@@ -256,9 +256,9 @@
 	done
 
 test-solver: test-solver.o solver_test.o verify.o matchfile.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 test-solver-2: test-solver-2.o solver_test_2.o verify.o matchfile.o quad-utils.o codefile.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 
 NODEP_OBJS += solver_test.o solver_test_2.o
 ALL_OBJ += test-solver.o test-solver-2.o
@@ -290,7 +290,7 @@
 # Add the dependencies here...
 TEST_DFIND_OBJS := dfind.o
 ALL_TEST_EXTRA_OBJS += $(TEST_DFIND_OBJS)
-test_dfind: $(TEST_DFIND_OBJS) $(ANUTILS_LIB) $(GSL_LIB)
+test_dfind: $(TEST_DFIND_OBJS) $(ANUTILS_LIB) 
 
 TEST_CTMF_OBJS := ctmf.o
 ALL_TEST_EXTRA_OBJS += $(TEST_CTMF_OBJS)
@@ -302,39 +302,39 @@
 
 test_multiindex2: test_multiindex2.o solver.o verify.o \
 	tweak2.o tweak.o matchfile.o matchobj.o quad-utils.o \
-	$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+	$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 ALL_TEST_EXTRA_OBJS += solver.o verify.o tweak2.o tweak.o quad-utils.o
 
 #../util/cutest.o \
 
-test_2mass: 2mass.o 2mass-fits.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_2mass: 2mass.o 2mass-fits.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += 2mass-fits.o 2mass.o
 
-test_usnob: usnob-fits.o usnob.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_usnob: usnob-fits.o usnob.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += usnob-fits.o usnob.o
 
-test_nomad: nomad-fits.o nomad.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_nomad: nomad-fits.o nomad.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += nomad-fits.o nomad.o
 
-test_matchfile: matchfile.o matchobj.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_matchfile: matchfile.o matchobj.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += matchfile.o matchobj.o
 
-test_blindutils: blindutils.o $(ANUTILS_LIB) $(GSL_LIB)
+test_blindutils: blindutils.o $(ANUTILS_LIB) 
 ALL_TEST_EXTRA_OBJS += blindutils.o
 
 test_dcen3x3: dcen3x3.o
 ALL_TEST_EXTRA_OBJS += dcen3x3.o
 
-test_simplexy: $(SIMPLEXY_OBJ) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_simplexy: $(SIMPLEXY_OBJ) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += $(SIMPLEXY_OBJ)
 
-test_resort-xylist: resort-xylist.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_resort-xylist: resort-xylist.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += resort-xylist.o
 
-test_tweak: test_tweak.o tweak.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+test_tweak: test_tweak.o tweak.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_TEST_EXTRA_OBJS += tweak.o
 
-ALL_TEST_LIBS += $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
+ALL_TEST_LIBS += $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) 
 
 include $(COMMON)/makefile.tests
 
@@ -349,43 +349,43 @@
 ######################################################
 
 demo_dsmooth: demo_dsmooth.o dsmooth.o \
-	$(COMMON)/cairoutils.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -pg $(LDLIBS) -o $@
+	$(COMMON)/cairoutils.o $(ANUTILS_LIB)  $(QFITS_LIB) 
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -pg $(GSL_LIB) $(LDLIBS) -o $@
 
 control-program: control-program.o \
 		$(BACKEND_OBJS) $(SIMPLEXY_OBJ) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
 ALL_OBJ += control-program.o
 
 # qfits is only needed for writing debugging images...
-image2xy: image2xy-main.o image2xy-files.o $(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
+image2xy: image2xy-main.o image2xy-files.o $(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += image2xy-main.o image2xy-files.o
 
-new-wcs: new-wcs-main.o new-wcs.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+new-wcs: new-wcs-main.o new-wcs.o $(ANUTILS_LIB)  $(QFITS_LIB)
 
 ALL_OBJ += new-wcs-main.o
 
 rdls2hpls: rdls2hpls.o \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 rdlstohealpix: rdlstohealpix.o \
-		$(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANUTILS_LIB)  $(QFITS_LIB)
 
 xylist2fits: xylist2fits.o \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 certifiable: certifiable.o matchfile.o matchobj.o solvedfile.o \
-		$(UTILS_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTILS_LIB) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 fits-guess-scale: fits-guess-scale-main.o fits-guess-scale.o \
-		$(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANUTILS_LIB)  $(QFITS_LIB)
 
 ALL_OBJ += fits-guess-scale-main.o
 
 get-wcs: get-wcs.o \
-		$(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANUTILS_LIB)  $(QFITS_LIB)
 
 
 ifdef WCSTOOLS_EXISTS
@@ -396,133 +396,133 @@
 WCS_LIB_DEP :=
 
 wcs-xy2rd: wcs-xy2rd-main.o wcs-xy2rd.o $(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) $(WCS_LIB_DEP)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-xy2rd-main.o
 
 wcs-rd2xy: wcs-rd2xy-main.o wcs-rd2xy.o $(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) $(WCS_LIB_DEP)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-rd2xy-main.o
 
 wcs-pv2sip: wcs-pv2sip.o tweak.o \
-		$(ANLIBS) $(GSL_LIB)
+		$(ANLIBS) 
 ALL_OBJ += wcs-pv2sip.o
 
-hpgrid: hpgrid.o $(ANUTILS_LIB) $(GSL_LIB)
+hpgrid: hpgrid.o $(ANUTILS_LIB) 
 
 $(CFITS_UTILS) :: %: %.o $(CFITS_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 fitsverify: ftverify.c fvrf_data.c fvrf_file.c fvrf_head.c fvrf_key.c fvrf_misc.c $(CFITS_DEP)
 	$(CC) -DSTANDALONE -trigraphs $(CFITS_INC) $^ $(CFITS_LIB) -lm -o $@
 
 $(OLDEXECS) :: %: %.o $(OLDEXECS_OBJS) \
-		$(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) $(QFITS_LIB)
+		$(ANUTILS_LIB)  $(LIBKD_LIB) $(QFITS_LIB)
 
-catalog_analysis: catalog_analysis.o pnpoly.o $(ANUTILS_LIB) $(GSL_LIB)
+catalog_analysis: catalog_analysis.o pnpoly.o $(ANUTILS_LIB) 
 
 nomadtofits: nomadtofits.o nomad-fits.o nomad.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 ucac3tofits: ucac3tofits.o ucac3-fits.o ucac3.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $^ $(LDFLAGS) -lbz2 $(LDLIBS) -o $@
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $^ $(LDFLAGS) -lbz2 $(GSL_LIB) $(LDLIBS) -o $@
 
 read_nomad: read_nomad.o nomad.o
 
-2masstofits: $(2MASSOBJ) $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(ZLIB_LIB) $(LDLIBS) -o $@
+2masstofits: $(2MASSOBJ) $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(ZLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 diffractionFlag_check: diffractionFlag_check.o usnob-fits.o usnob.o \
-		$(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 project-usnob: project-usnob.o usnob-fits.o usnob.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 usnobtofits: usnobtofits.o usnob-fits.o usnob.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
-hpowned: hpowned.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+hpowned: hpowned.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 build-an-catalog: build-an-catalog.o usnob-fits.o usnob.o \
 		2mass-fits.o 2mass.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 cut-an: cut-an.o catalog.o usnob.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 startree: startree.o catalog.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 startree2: startree2-main.o startree2.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += startree2-main.o startree2.o
 
 uniformize-catalog: uniformize-catalog-main.o uniformize-catalog.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += uniformize-catalog-main.o uniformize-catalog.o
 
 build-index: build-index-main.o build-index.o uniformize-catalog.o startree2.o hpquads.o \
 		quad-builder.o quad-utils.o codefile.o codetree.o unpermute-stars.o \
 		unpermute-quads.o merge-index.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += build-index-main.o build-index.o uniformize-catalog.o quad-utils.o
 
 hpquads: hpquads-main.o hpquads.o quad-builder.o quad-utils.o codefile.o pnpoly.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += hpquads-main.o quad-builder.o hpquads.o
 
 allquads: allquads-main.o allquads.o quad-builder.o quad-utils.o codefile.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 codetree: codetree-main.o codetree.o codefile.o quad-utils.o \
-		$(UTIL_OBJS) $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += codetree-main.o
 
 unpermute-stars: unpermute-stars-main.o unpermute-stars.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += unpermute-stars-main.o
 
 unpermute-quads: unpermute-quads-main.o unpermute-quads.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += unpermute-quads-main.o
 
 merge-index: merge-index-main.o merge-index.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 ALL_OBJ += merge-index-main.o
 
 local-index: local-index.o build-index.o codetree.o startree2.o codetree.o \
 		hpquads.o uniformize-catalog.o merge-index.o unpermute-quads.o unpermute-stars.o \
 		quad-builder.o quad-utils.o codefile.o wcs-xy2rd.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(LDLIBS) -o $@
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += local-index.o
 
 index-to-table: index-to-table.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 threadtest: threadtest.o $(BACKEND_LIB) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
 
 backend: backend-main.o $(BACKEND_LIB) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
 ### TWEAK_DEBUG_PLOTS
 #backend: backend-main.o $(BACKEND_LIB) \
-#		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB) \
+#		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB)  \
 #		$(PLOTSTUFF) $(COMMON)/cairoutils.o
 #	$(CC) -o $@ $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB)
 
 verify: verify-main.o verify.o verify2.o matchfile.o matchobj.o $(PLOTSTUFF) $(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
 	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) -o $@
 
 keirthing: keirthing.o tweak.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 
 dstnthing: dstnthing.o tweak2.o tweak.o matchfile.o matchobj.o verify.o scamp.o $(PLOTSTUFF) \
-		$(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB) 
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
+		$(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB)  
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += dstnthing.o
 
 ALL_OBJ += verify-main.o
@@ -531,15 +531,15 @@
 		image2xy-files.o resort-xylist.o tabsort.o cut-table.o \
 		fits-guess-scale.o wcs-rd2xy.o new-wcs.o \
 		scamp.o \
-		$(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(CFITS_LIB) $(LDLIBS) -o $@
+		$(SIMPLEXY_OBJ) $(CFITS_DEP) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) $(CFITS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += scamp.o cut-table.o
 
 augment-xylist: augment-xylist-main.o augment-xylist.o tabsort.o cut-table.o fits-guess-scale.o \
 		image2xy-files.o resort-xylist.o blindutils.o \
 		$(SIMPLEXY_OBJ) $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(CFITS_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CFITS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 ALL_OBJ += augment-xylist-main.o
 
@@ -547,21 +547,21 @@
 		solvedclient.o solvedfile.o tweak2.o verify.o \
 		tweak.o \
 		$(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(LDLIBS) -o $@.tmp
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(GSL_LIB) $(LDLIBS) -o $@.tmp
 	mv $@.tmp $@
 
 tweak: tweak-main.o verify.o matchfile.o \
 		$(COMMON)/cairoutils.o $(PLOTSTUFF) \
 		$(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) 
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 whynot: whynot.o matchfile.o matchobj.o codefile.o solvedclient.o \
 		solvedfile.o verify.o quad-utils.o \
 		$(PLOTSTUFF) $(COMMON)/cairoutils.o $(CATS_LIB) \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB) $(GSL_LIB) 
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB)  
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += whynot.o
 
 quadcenters: quadcenters.o \
@@ -571,36 +571,36 @@
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(LIBKD_LIB) $(QFITS_LIB)
 
 agreeable: agreeable.o matchfile.o matchobj.o solvedfile.o solvedclient.o verify.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(LIBKD_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) $(LIBKD_LIB) $(QFITS_LIB)
 
-rdlsinfo: rdlsinfo.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+rdlsinfo: rdlsinfo.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
-xylsinfo: xylsinfo.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+xylsinfo: xylsinfo.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 wcsinfo: wcsinfo.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) \
-		$(QFITS_LIB) $(GSL_LIB)
+		$(QFITS_LIB) 
 
-index-info: index-info.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+index-info: index-info.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 subwcs: subwcs.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) \
-		$(QFITS_LIB) $(GSL_LIB)
+		$(QFITS_LIB) 
 ALL_OBJS += subwcs.o
 
-wcs-grab: wcs-grab.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) \
+wcs-grab: wcs-grab.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  \
 		$(QFITS_LIB)
 
-printsolved: printsolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
+printsolved: printsolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB) 
 
-setsolved: setsolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
+setsolved: setsolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB) 
 
-mergesolved: mergesolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+mergesolved: mergesolved.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
-solvedserver: solvedserver.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) \
+solvedserver: solvedserver.o solvedfile.o $(UTIL_OBJS) $(ANUTILS_LIB)  \
 		$(QFITS_LIB)
-	$(CC) $(CFLAGS) $^ $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(GSL_LIB) $(LDLIBS) -o $@
 
 plotcat: plotcat.o 2mass-fits.o 2mass.o nomad-fits.o nomad.o catalog.o usnob-fits.o usnob.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(LIBKD_LIB) \
 		$(QFITS_LIB)
 
 plotquad.o: plotquad.c 
@@ -610,28 +610,28 @@
 	$(MAKE) -C $(COMMON) cairoutils.o
 
 onepixel: onepixel.o $(COMMON)/cairoutils.o
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 plotquad: plotquad.o matchfile.o matchobj.o $(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) 
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 plot-xy-and-quad: plot-xy-and-quad.o matchfile.o $(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) 
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) 
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 plot-constellations.o: plot-constellations.c
 	$(CC) $(CFLAGS) $(CAIRO_INC) $(NETPBM_INC) $(CATS_INC) -c -o $@ $<
 
 plot-constellations: plot-constellations.o $(CATS_LIB) $(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 catalog-dists: catalog-dists.o 2mass-fits.o usnob-fits.o usnob.o \
-		$(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
-add-text: add-text.o $(COMMON)/cairoutils.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+add-text: add-text.o $(COMMON)/cairoutils.o $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $(LDFLAGS) $^ $(CAIRO_LIB) $(NETPBM_LIB) $(PNG_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 PYTHON_EXECS := plotann.py
 PYTHON_INSTALL := $(PYTHON_EXECS)
@@ -644,8 +644,8 @@
 plotstuff.o: plotstuff.c
 	$(CC) $(CFLAGS) $(CAIRO_INC) $(NETPBM_INC) -c -o $@ $<
 
-plotstuff: plotstuff-main.o $(PLOTSTUFF) $(CATS_LIB) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
+plotstuff: plotstuff-main.o $(PLOTSTUFF) $(CATS_LIB) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += plotstuff-main.o
 
 PLOTSTUFF_SRCS = $(subst .o,.c,$(PLOTSTUFF))
@@ -662,7 +662,7 @@
 
 _plotstuff_c.so: plotstuff_wrap.c setup.py $(PLOTSTUFF) $(CATS_LIB) \
 		$(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 	python setup.py build --force --build-base build --build-platlib build/lib
 	cp build/lib/_plotstuff_c.so $@
 
@@ -673,59 +673,59 @@
 	$(CC) $(CFLAGS) $(CAIRO_INC) $(NETPBM_INC) -c $< -o $@
 
 plotxy: plotxy-main.o $(PLOTSTUFF) $(CATS_LIB) $(COMMON)/cairoutils.o \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LDLIBS) -o $@
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 ALL_OBJ += plotxy-main.o
 
-test_plotstuff: test_plotstuff-main.o test_plotstuff.o $(COMMON)/cutest.o $(PLOTSTUFF) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(LDLIBS) -o $@
+test_plotstuff: test_plotstuff-main.o test_plotstuff.o $(COMMON)/cutest.o $(PLOTSTUFF) $(COMMON)/cairoutils.o $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(WCSLIB_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 test_plotstuff-main.c: test_plotstuff.c
 	$(AN_SHELL) $(MAKE_TESTS) $^ > $@
 
 query-starkd: query-starkd.o $(ANLIBS)
 ALL_OBJ += query-starkd.o
 
-quadidx: quadidx.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+quadidx: quadidx.o $(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 codeprojections: codeprojections.o $(UTIL_OBJS) $(ANLIBS)
 
 quadscales: quadscales.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(LIBKD_LIB) \
 		$(QFITS_LIB)
 
 index-stats: index-stats.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(LIBKD_LIB) \
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(LIBKD_LIB) \
 		$(QFITS_LIB)
 
 quadsperstar: quadsperstar.o \
-		$(UTIL_OBJS) $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) \
+		$(UTIL_OBJS) $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB)  \
 		$(QFITS_LIB)
 
-fitsgetext: fitsgetext.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+fitsgetext: fitsgetext.o $(ANUTILS_LIB)  $(QFITS_LIB)
 
-fits-dedup: fits-dedup.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+fits-dedup: fits-dedup.o $(ANUTILS_LIB)  $(QFITS_LIB)
 
-subtable: subtable.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+subtable: subtable.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
-tabsort: tabsort-main.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+tabsort: tabsort-main.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 ALL_OBJ += tabsort-main.o
 
-column-merge: column-merge.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+column-merge: column-merge.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 resort-xylist: resort-xylist-main.o resort-xylist.o $(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 ALL_OBJ += resort-xylist-main.o
 
-bgsubtract: bgsubtract.o ctmf.o $(COMMON)/cairoutils.o $(UTIL_OBJS) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
-	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(LDLIBS) -o $@
+bgsubtract: bgsubtract.o ctmf.o $(COMMON)/cairoutils.o $(UTIL_OBJS) $(ANUTILS_LIB)  $(QFITS_LIB)
+	$(CC) $^ $(LDFLAGS) $(CAIRO_LIB) $(PNG_LIB) $(NETPBM_LIB) $(JPEG_LIB) $(GSL_LIB) $(LDLIBS) -o $@
 
 usnob-scamp-catalog: usnob-scamp-catalog.o usnob-fits.o usnob.o $(UTIL_OBJS) \
-		$(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 create-scamp-catalog: create-scamp-catalog.o \
-		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB)
+		$(UTIL_OBJS) $(ANFILES_LIB) $(ANUTILS_LIB)  $(QFITS_LIB)
 
 DEP_OBJ := $(ALL_OBJ)
 DEP_PREREQS := $(QFITS_LIB)
diff -Nur astrometry.net-0.43.as-needed/Makefile astrometry.net-0.43/Makefile
--- astrometry.net-0.43.as-needed/Makefile	2013-06-13 15:54:15.934430711 -0700
+++ astrometry.net-0.43/Makefile	2013-06-13 15:56:14.795089358 -0700
@@ -67,7 +67,7 @@
 	$(MAKE) -C qfits-an/src
 
 gsl-an:
-	$(MAKE) -C gsl-an
+	echo "nothing to do"
 
 .PHONY: subdirs thirdparty qfits-an gsl-an
 
diff -Nur astrometry.net-0.43.as-needed/util/Makefile astrometry.net-0.43/util/Makefile
--- astrometry.net-0.43.as-needed/util/Makefile	2013-06-13 17:05:10.551069687 -0700
+++ astrometry.net-0.43/util/Makefile	2013-06-13 17:10:31.830861262 -0700
@@ -181,7 +181,7 @@
 
 resample: resample-main.o \
 		$(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(GSL_LIB) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += resampe.o
 
 tycho2tofits: tycho2tofits.o \
@@ -194,18 +194,18 @@
 
 _sip.so: sip-utils.o gslutils.o sip.o starutil.o mathutil.o sip_qfits.o \
 		fitsioutils.o errors.o ioutils.o bl.o os-features.o an-endian.o \
-		log.o tic.o $(QFITS_LIB) $(GSL_LIB)
-	$(CC) $(SHAREDLIBFLAGS) $^ $(GSL_LIB) -o $@
+		log.o tic.o $(QFITS_LIB) 
+	$(CC) $(LDFLAGS) $(SHAREDLIBFLAGS) $^ $(GSL_LIB) -o $@
 
 _healpix.so: healpix-utils.o healpix.o permutedsort.o starutil.o mathutil.o errors.o ioutils.o bl.o log.o tic.o os-features.o
-	$(CC) $(SHAREDLIBFLAGS) $^ -o $@
+	$(CC) $(LDFLAGS) $(SHAREDLIBFLAGS) $^ $(GSL_LIB) -o $@
 
 util.py util_wrap.c: util.i $(ANUTILS_H)
 	swig -python -I. $<
 
 _util.so: util_wrap.c setup-util.py \
-		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
-	WCSLIB_LIB="$(WCSLIB_LIB)" GSL_LIB="$(GSL_LIB)" \
+		$(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
+	WCSLIB_LIB="$(WCSLIB_LIB)" GSL_LIB="" \
 	python setup-util.py build --force --build-base build --build-platlib build/lib
 	cp build/lib/_util.so $@
 
@@ -224,23 +224,23 @@
 	cp build/lib/_index_c.so _index_c.so
 	cp build/lib/_index_util.so _index_util.so
 
-coadd: coadd-main.o $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB) $(WCS_LIB_DEP)
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCS_LIB) $(LDLIBS) -o $@
+coadd: coadd-main.o $(ANUTILS_LIB) $(QFITS_LIB)  $(WCS_LIB_DEP)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(GSL_LIB) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += coadd.o coadd-main.o
 
-wcs-resample: wcs-resample-main.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
+wcs-resample: wcs-resample-main.o $(ANUTILS_LIB)  $(QFITS_LIB) $(WCS_LIB_DEP)
+	$(CC) $(LDFLAGS) $^ $(GSL_LIB) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-resample-main.o
 
-fit-wcs: fit-wcs-main.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
+fit-wcs: fit-wcs-main.o $(ANFILES_LIB) $(LIBKD_LIB) $(ANUTILS_LIB)  $(QFITS_LIB) $(WCS_LIB_DEP)
+	$(CC) $^ $(LDFLAGS) $(GSL_LIB) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += fit-wcs-main.o
 
-get-healpix: get-healpix.o $(ANUTILS_LIB) $(GSL_LIB)
+get-healpix: get-healpix.o $(ANUTILS_LIB) 
 ALL_OBJ += get-healpix.o
 
-wcs-to-tan: wcs-to-tan.o $(ANUTILS_LIB) $(GSL_LIB) $(QFITS_LIB) $(WCS_LIB_DEP)
-	$(CC) $^ $(LDFLAGS) $(WCS_LIB) $(LDLIBS) -o $@
+wcs-to-tan: wcs-to-tan.o $(ANUTILS_LIB)  $(QFITS_LIB) $(WCS_LIB_DEP)
+	$(CC) $^ $(LDFLAGS) $(GSL_LIB) $(WCS_LIB) $(LDLIBS) -o $@
 ALL_OBJ += wcs-to-tan.o
 
 pad-file: pad-file.o $(ANUTILS_LIB)
@@ -482,7 +482,7 @@
 	test_convolve_image test_qsort_r test_wcs test_big_tables
 # test_hd depends on hd.fits...
 ALL_TEST_EXTRA_OBJS = 
-ALL_TEST_LIBS = $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+ALL_TEST_LIBS = $(LIBKD_LIB) $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 ALL_TEST_EXTRA_LDFLAGS = $(WCSLIB_LIB)
 
 include $(COMMON)/makefile.tests
@@ -499,12 +499,12 @@
 
 test_errors: $(ANUTILS_LIB)
 
-test_sip-utils: $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+test_sip-utils: $(ANUTILS_LIB) $(QFITS_LIB) 
 
-test_anwcs: $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+test_anwcs: $(ANUTILS_LIB) $(QFITS_LIB) 
 	$(CC)$(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) -o $@
 
-test_wcs: $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) $(GSL_LIB)
+test_wcs: $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB) 
 	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(WCSLIB_LIB) -o $@
 
 test_tycho2: $(ANFILES_LIB) $(ANUTILS_LIB) $(QFITS_LIB)
@@ -537,7 +537,7 @@
 
 test_starutil: $(ANUTILS_LIB)
 
-test_svd: $(ANUTILS_LIB) $(GSL_LIB)
+test_svd: $(ANUTILS_LIB) 
 
 ### /TESTS
 
diff -Nur astrometry.net-0.43.as-needed/util/makefile.gsl astrometry.net-0.43/util/makefile.gsl
--- astrometry.net-0.43.as-needed/util/makefile.gsl	2013-06-13 15:54:15.937430729 -0700
+++ astrometry.net-0.43/util/makefile.gsl	2013-06-13 15:54:33.183525063 -0700
@@ -14,10 +14,5 @@
 # along with the Astrometry.net suite ; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
 
-GSL_DIR := $(BASEDIR)/gsl-an
-GSL_INC := -I$(GSL_DIR)
-GSL_LIB := $(GSL_DIR)/libgsl-an.a
-
-$(GSL_LIB):
-	$(MAKE) -C $(GSL_DIR)
-
+GSL_INC ?= $(shell pkg-config --cflags gsl 2>/dev/null)
+GSL_LIB ?= $(shell pkg-config --libs gsl 2>/dev/null)
